ui/core/components/suggest_reforges_action.tsx

```diff
@@ -1164,7 +1169,15 @@ export class ReforgeOptimizer {
 			const filteredGemDataForColor = new Array<GemData>();
 
 			for (const gem of allGemsOfColor) {
-				if (gem.requiredProfession > 0 || gem.name.includes('Perfect') || !gemMatchesSocket(gem, socketColor)) {
+				if (
+					(gem.requiredProfession == Profession.Jewelcrafting && !this.player.hasProfession(Profession.Jewelcrafting)) ||
+					// We check explicitly for tank specs OR non-tanks and primary stats to reduce the number of combinations
+					// this is fine for now as primary stat budgets relatively are higher
```

Comment:
Slightly clearer comment: "Force non-tank specs to use exclusively primary stat JC gems to speed up calculations. May need to revisit this approximation at higher ilvls."